server/api/api/endpoints/runs.py

```diff
@@ -502,3 +502,57 @@ async def abort_run(
     )
 
     return background_task
+
+
+@router.get("/projects/{project}/completed_runs")
+async def list_completed_runs(
+    project: str = None,
+    sort: bool = True,
+    iter: bool = True,
+    start_time_from: str = None,
+    start_time_to: str = None,
+    page: int = Query(None, gt=0),
+    page_size: int = Query(None, alias="page-size", gt=0),
+    page_token: str = Query(None, alias="page-token"),
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    if project != "*":
+        await server.api.utils.auth.verifier.AuthVerifier().query_project_permissions(
+            project,
+            mlrun.common.schemas.AuthorizationAction.read,
+            auth_info,
+        )
+
+    paginator = server.api.utils.pagination.Paginator()
+
+    async def _filter_runs(_runs):
```

Comment:
This function is duplicated from `list_runs`. maybe refactoring it into a private method and using it in both places.